Sources/Player/OfflineEngine/Internal/Storage/DBStorage/DBOfflineStorage.swift

```diff
@@ -0,0 +1,103 @@
+import Foundation
+import GRDB
+
+// MARK: - DBOfflineStorage
+
+class DBOfflineStorage {
+	private let dbQueue: DatabaseQueue
+
+	init(dbQueue: DatabaseQueue) {
+		self.dbQueue = dbQueue
+		try? initializeDatabase()
+	}
+}
+
+// MARK: OfflineStorage
+
+extension DBOfflineStorage: OfflineStorage {
+	// MARK: - Save OfflineEntry
+
+	func save(_ entry: OfflineEntry) throws {
+		let entryDTO = DBOfflineEntryDTO(from: entry)
+		try dbQueue.write { db in
+			try entryDTO.insert(db)
+		}
+	}
+
+	// MARK: - Get OfflineEntry by MediaProduct
+
+	func get(mediaProduct: MediaProduct) throws -> OfflineEntry? {
```

Comment:
[question] to align with approach from previous PR and be more consistent, where we inform only the key, maybe here we do the same? Or change in the first one? Then it's either responsibility of this func to get the key or of the caller to inform it.
(I'm not adding comment, but it's also valid for the func below.)